besu/src/main/java/org/hyperledger/besu/controller/BesuController.java

```diff
@@ -191,6 +192,21 @@ BesuControllerBuilder fromGenesisConfig(
           genesisConfig.getConfigOptions(genesisConfigOverrides);
       final BesuControllerBuilder builder;
 
+      // TODO-lucas Too much in-line code?
+      if (configOptions.isQbft() && (configOptions.isIbft2() || configOptions.isIbftLegacy())) {
```

Comment:
may be add a comment mentioning something on the lines of `detect migration from ibft2/ibft to qbft...`

Reply:
I don't think it looks too unreadable as it is. However, if we think it's too much I'd vote for an extracted method rather than a comment, e.g. `detectMigrationFromIbftToQbft()`

Although is that comment/name accurate? Are we able to detect that it's _from_ ibft and _to_ qbft?

Reply:
Just realised you're talking about the whole if block, not just the condition :)
Still, I don't think think it's too bad. 
One to bear in mind for refactoring nearer the completion of the PR maybe, in case it changes.

Reply:
A seperate function would be nice for this